language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ClassForType.scala

```diff
@@ -18,67 +17,94 @@ object ClassForType extends StrictLogging {
       packagePrefixes: Map[PackageId, String],
   ): List[JavaFile] = {
 
-    val className =
-      ClassName.bestGuess(fullyQualifiedName(typeWithContext.identifier, packagePrefixes))
-    val javaPackage = className.packageName()
+    val classNameString = fullyQualifiedName(typeWithContext.identifier, packagePrefixes)
+    val className = ClassName.bestGuess(classNameString)
 
-    typeWithContext.`type`.typ match {
+    def generate(lfInterfaceType: InterfaceType): List[JavaFile] =
+      generateInterfaceTypes(typeWithContext, packagePrefixes) ++
```

Comment:
I moved the interface generation here alongside everything else. It seemed out of place out of here. Originally I tried to generate this as part of the generation of a template but some generated code ended up missing, probably because I wasn't considering interfaces implementing interfaces. If that's the case, we may want to make the generation of interfaces recursive so that we can isolate in the generation of templates.